### ISSUE TITLE
Possibly flawed implementation around UseStaticFiles

### ISSUE THREAD

**Issue by taori** [dotnet/aspnetcore#34159]
### Describe the bug
I am using UseStaticFiles to provide additional content types in my blazor application (".apk","application/vnd.android.package-archive")
However as soon as i use the overload blazor.server.js will be 404. If i revert back to the overload without parameters the blazor.server.js loads just fine. 

### To Reproduce

```cs
	// var contentTypeProvider = new FileExtensionContentTypeProvider();
	// // contentTypeProvider.Mappings.Add(".apk", "application/vnd.android.package-archive");
	// contentTypeProvider.Mappings[".apk"] = "application/vnd.android.package-archive";
	// app.UseStaticFiles(new StaticFileOptions(){ ContentTypeProvider = contentTypeProvider});
	app.UseStaticFiles();
```

### Exceptions (if any)
none

### Further technical details
- ASP.NET Core version net5.0
.NET SDK (gemäß "global.json"):
 Version:   5.0.301
 Commit:    ef17233f86

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.301\

Host (useful for support):
  Version: 5.0.7
  Commit:  556582d964

.NET SDKs installed:
  5.0.301 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version
VS 2019 Version 16.10.2

**Comment by taori** [comment]
Additional information:
If i do this:

```cs
	app.UseHttpsRedirection();
	var contentTypeProvider = new FileExtensionContentTypeProvider();
	// contentTypeProvider.Mappings.Add(".apk", "application/vnd.android.package-archive");
	contentTypeProvider.Mappings[".apk"] = "application/vnd.android.package-archive";
	app.UseStaticFiles(new StaticFileOptions(){ ContentTypeProvider = contentTypeProvider});
	app.UseStaticFiles();
```
blazor.server.js will load, apk download will work too, but the page will say "sorry, there is nothing at this address."

So the page behavior is weird (because the download works, but the page missbehaves and also the code is very unordinary (since there is nothing else in the Startup.Configure method where i have to call UseXXX multiple times for things to work

**Comment by Tratcher** [comment]
See https://github.com/dotnet/aspnetcore/issues/25354#issuecomment-712942116

**Comment by taori** [comment]
Doing it as described in that other issue i can get both blazor.js+apk download working with just UseStaticFiles() - Perhaps this should be looked at then? I don't think how i configured it would be unreasonable given the current API. If this works in a more reliable fashion when using Configure<StaticFileOptions> perhaps the UseXXX overload should be removed and the comment should point out to use configure instead?

**Comment by Tratcher** [comment]
This is limitation specific to the blazor design and I agree there's room for usability improvements.

**Comment by javiercn** [comment]
This was fixed in 8.0.

Blazor no longer messes up with the default file options
